pkg/exchange/binance/binanceapi/client.go

```diff
@@ -140,13 +155,37 @@ func (c *RestClient) SendRequest(req *http.Request) (*requestgen.Response, error
 }
 
 // newAuthenticatedRequest creates new http request for authenticated routes.
-func (c *RestClient) NewAuthenticatedRequest(ctx context.Context, method, refURL string, params url.Values, payload interface{}) (*http.Request, error) {
-	if len(c.Key) == 0 {
-		return nil, errors.New("empty api key")
+func (c *RestClient) NewAuthenticatedRequest(
+	ctx context.Context, method, refURL string, params url.Values, payload interface{},
+) (*http.Request, error) {
+	signFunc := c.signHMAC
+	if c.IsUsingEd25519Auth() {
+		signFunc = c.signEd25519
 	}
+	return c.newAuthenticatedRequest(ctx, method, refURL, params, payload, signFunc)
+}
+
+// newAuthenticatedRequest creates new http request for authenticated routes.
+// See the doc links for details:
+// - https://developers.binance.com/docs/binance-spot-api-docs/rest-api/endpoint-security-type
+// - https://developers.binance.com/docs/binance-spot-api-docs/rest-api/general-information-on-endpoints
+func (c *RestClient) newAuthenticatedRequest(
+	ctx context.Context, method, refURL string, params url.Values, payload interface{}, signFunc func(string) string,
+) (*http.Request, error) {
+	if c.IsUsingEd25519Auth() {
+		// Ed25519 authentication
+		if len(c.PrivateKey) == 0 {
+			return nil, errEmptyPrivateKey
```

Comment:
It should be impossible to happen while you have passed `IsUsingEd25519Auth`.

Reply:
I just updated the validations